### PR TITLE
Feature/build improvements

### PR DESCRIPTION
**Pull request by neophob**

### CODE REVIEW

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -20,6 +20,8 @@ jobs:
     - name: npm install, build, and test
       run: |
         npm ci
-        npm test
+        TS_NODE_TRANSPILE_ONLY=true time npm test
```
Comment (neophob) [inline .github/workflows/nodejs.yml]: this is the fix, found in https://github.com/TypeStrong/ts-node/issues/754.

ideally we should update tsconfig.json

File: tsconfig.json [hunk 2]
```diff
@@ -8,9 +8,12 @@
 		"allowJs": true
 	},
 	"exclude": [
+		".nyc_output",
 		"node_modules",
 		"dist",
-		"test",
+		"coverage",
+		"res",
+		"test/fixtures",
```
Comment (neophob) [inline tsconfig.json]: note: added the root test dir too, as we have ts files in there

**Comment by freezy** [comment]
I'm getting this on Windows:

```
> vpx-js@1.2.5 test C:\Development\vpx-js
> npm run compile && nyc npx mocha


> vpx-js@1.2.5 compile C:\Development\vpx-js
> nearleyc lib/scripting/vbscript.ne -o lib/scripting/vbscript.ts

UNCAUGHT_EXCEPTION { error: '⨯ Unable to compile TypeScript:\n' }

=============================== Coverage summary ===============================
Statements   : Unknown% ( 0/0 )
Branches     : Unknown% ( 0/0 )
Functions    : Unknown% ( 0/0 )
Lines        : Unknown% ( 0/0 )
================================================================================
```

File: .github/workflows/nodejs.yml [hunk 3]
```diff
@@ -20,6 +20,8 @@ jobs:
     - name: npm install, build, and test
       run: |
         npm ci
-        npm test
+        time npm test
+        npm run lint || true
```
Comment (neophob) [inline .github/workflows/nodejs.yml]: can be enabled (fail when lint fail) once the test directory is fixed (there are some lint errors)

Comment (freezy) [reply .github/workflows/nodejs.yml]: Isn't there a separate GitHub app that lints? Would be nice to have a separate check for this..

**Pull request merged**